app/backend/src/RPG/service/RPGService.ts

```diff
@@ -1,38 +1,37 @@
-import Service, { ServiceError } from '../../abstractClasses/service'
-import RPG, { rpgZodSchema }  from '../schemas/RPG'
-import RPGModel from '../model/RPGModel'
+import Service, { ServiceError } from '../../abstractClasses/service';
+import RPG, { rpgZodSchema }  from '../schemas/RPG';
+import RPGModel from '../model/RpgModel';
 
 class RPGService extends Service<RPG> {
   constructor(protected model: RPGModel) {
-    super(model)
+    super(model);
   }
 
   create = async(obj: RPG): Promise<RPG | null | ServiceError> => {
     const parsed = rpgZodSchema.safeParse(obj);
```

Comment:
this looks really cool, but I think we can save time on adding the zod validation on the middleware  on the routes, like, lets not worry at all instantiating this in case the request is not well formed and Zod along with Express can handle this and send it back to the user.